src/records.js

```diff
@@ -28,10 +28,35 @@ const addRecord = async (recordsContract, dataHash, metadata, dataUri, ethParams
   } catch (e) {
     if (e.message === 'sender account not recognized') {
       throw new Error('The web3 Instance that you pass to Linnia cannot sign a transaction for this address');
+    } else {
+      throw new Error('Something went wrong');
     }
   }
+};
+
+const signRecord = async (recordsContract, usersContract, dataHash, ethParams) => {
+  // Check provenance of attestator
+  const provenance = await usersContract.provenanceOf(ethParams.from);
+  if (!(provenance > 0)) {
+    throw new Error('The attestor does not have provenance (Invalid Attestator)');
+  }
 
-  return undefined;
+  // Check if attestator have signed the record already
+  const sigExists = await recordsContract.sigExists(dataHash, ethParams.from);
+  if (sigExists) {
+    throw new Error('The attestor have already signed this record');
+  }
+
```

Comment:
Can we add basic validation of the datahash?

Reply:
Sure!